File: 06_serialization/src/main/cpp/dataset.cpp
#include "dataset.h"

#include <iostream>
MeasurementInfo Dataset::getInfo() const
{
    return info;
}

void Dataset::setInfo(const MeasurementInfo &value)
{
    info = value;
}

std::map<DataType, std::vector<double> > Dataset::getRecords() const
{
    return records;
}

void Dataset::setRecords(const std::map<DataType, std::vector<double> > &value)
{
    records = value;
}

Dataset::Dataset()
{

}

void Dataset::Serialize(Json::Value &root)
{

}

void Dataset::Deserialize(Json::Value &root)
{
    info.Deserialize(root["info"]);
    const Json::Value& recordsJ = root["records"];
    for ( int typeI = DOWNLOAD; typeI <= PING; typeI++ )
    {
        DataType typeE = static_cast<DataType>(typeI);
        std::string typeS;
        switch (typeE) {
        case DOWNLOAD:
            typeS = "DOWNLOAD";
            break;
        case UPLOAD:
            typeS = "UPLOAD";
            break;
        case PING:
            typeS = "PING";
            break;
        default:
            typeS = "";
            break;
        }

        std::vector<double> data;
        const Json::Value& typeV = recordsJ[typeS];
        for (int i = 0; i < typeV.size(); i++){
            data.push_back(typeV[i].asDouble());
        }

        records.insert(std::make_pair(typeE,data));
    }
}

File: 06_serialization/src/main/java/cz/esw/serialization/json/Dataset.java
package cz.esw.serialization.json;

import java.util.List;
import java.util.Map;

/**
 * @author Marek Cuchý
 */
public class Dataset {

    private MeasurementInfo info;
    private Map<DataType, List<Double>> records;

    public Dataset() {
    }

    public Dataset(MeasurementInfo info, Map<DataType, List<Double>> records) {
        this.info = info;
        this.records = records;
    }

    public MeasurementInfo getInfo() {
        return info;
    }

    public void setInfo(MeasurementInfo info) {
        this.info = info;
    }

    public Map<DataType, List<Double>> getRecords() {
        return records;
    }

    public void setRecords(Map<DataType, List<Double>> records) {
        this.records = records;
    }
}
